napari/layers/vectors/vectors.py

```diff
@@ -242,9 +242,8 @@ def __init__(
 
         # Data containing vectors in the currently viewed slice
         self._view_data = np.empty((0, 2, 2))
-        self._displayed_stored = []
-        self._view_indices = []
-        self._view_alphas = []
+        self._view_indices: List[int] = []
```

Comment:
Technically this is `Union[np.ndarray, List[int]]` because `_set_view_slice` can assign it to the result of `np.arange`.

But given this is for indexing, I like the simplicity of `List[int]` (over `np.ndarray`), so it would be great just to coerce that particular value to a list.

Reply:
> But given this is for indexing, I like the simplicity of List[int] (over np.ndarray), so it would be great just to coerce that particular value to a list.

Done

Reply:
Sorry to mess you about here, but seeing the `list(range(...` call makes me realize there is a performance penalty here for a case that should be fast - i.e. for 2D points.

So I think it's better to revert to `np.arange` there and just make the typing more complicated with the `Union` - i.e. in the spirit of adding typing and making minimal changes to the code at runtime.

Of course `slice(...)` would be much faster and preferred over `np.arange` for that case and now I'm seeing all kinds of related small optimizations. I'd be happy to make a PR on top of this with those, but I could easily do that as a follow up instead.

Reply:
If you'd like to PR on top of this and push to my branch go ahead - alternatively let me know if not and I can revert back to `np.arange` and fix the typing appropriately.

Reply:
For now, if you can revert to `arange` and update the typing, that would be great. I like just clarifying the typing with minimal changes, then following up with optimizations and improvements later once the type checks are actually running.